### PR TITLE
Avoid reconciling `SriovNetworkNodePolicies` multiple times

### PR DESCRIPTION
**Pull request by zeeke**
`SriovNetworkNodePolicy` reconcile function lists all node policies and updates all SriovNetworkNodeState. Hence there is no need to resync all of them periodically.

Avoid triggering multiple reconciliations when creating multiple policies in a row (that is what happens when using yaml files).

### CODE REVIEW

File: controllers/sriovnetworknodepolicy_controller.go [hunk 1]
```diff
@@ -152,8 +163,29 @@ func (r *SriovNetworkNodePolicyReconciler) Reconcile(ctx context.Context, req ct
...
+
+	squashAndDebounceHandler := handler.Funcs{
+		CreateFunc: func(e event.CreateEvent, q workqueue.RateLimitingInterface) {
+			log.Log.WithName("SriovNetworkNodePolicy").
+				Info("Enqueuing sync for create event", "resource", e.Object.GetName())
+			qHandler(q)
+		},
+		UpdateFunc: func(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
+			log.Log.WithName("SriovNetworkNodePolicy").
+				Info("Enqueuing sync for create event", "resource", e.ObjectNew.GetName())
```
Comment (SchSeba) [inline controllers/sriovnetworknodepolicy_controller.go:181]: nit: create -> update

**Review by SchSeba — COMMENTED** [review]
general question is it possible for the user to create an object with that name?

can we use a non existing namespace or something like that to be sure there is not going to be an object like that?

File: controllers/sriovnetworknodepolicy_controller.go [hunk 2]
```diff
@@ -152,8 +163,29 @@ func (r *SriovNetworkNodePolicyReconciler) Reconcile(ctx context.Context, req ct
...
 func (r *SriovNetworkNodePolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	qHandler := func(q workqueue.RateLimitingInterface) {
+		q.AddAfter(reconcile.Request{NamespacedName: types.NamespacedName{
+			Namespace: namespace,
+			Name:      nodePolicySyncEventName,
+		}}, time.Second)
+	}
+
+	squashAndDebounceHandler := handler.Funcs{
+		CreateFunc: func(e event.CreateEvent, q workqueue.RateLimitingInterface) {
```
Comment (e0ne) [inline controllers/sriovnetworknodepolicy_controller.go:174]: Why we don't have DeleteFunc here?

File: controllers/sriovnetworknodepolicy_controller.go [hunk 3]
```diff
@@ -152,8 +163,29 @@ func (r *SriovNetworkNodePolicyReconciler) Reconcile(ctx context.Context, req ct
...
 // SetupWithManager sets up the controller with the Manager.
 func (r *SriovNetworkNodePolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	qHandler := func(q workqueue.RateLimitingInterface) {
+		q.AddAfter(reconcile.Request{NamespacedName: types.NamespacedName{
+			Namespace: namespace,
+			Name:      nodePolicySyncEventName,
+		}}, time.Second)
+	}
+
+	squashAndDebounceHandler := handler.Funcs{
```
Comment (adrianchiris) [inline controllers/sriovnetworknodepolicy_controller.go:173]: nit: just name it `delayedEventHandler` ?

File: controllers/sriovnetworknodepolicy_controller.go (hunk 2 above)
Comment (adrianchiris) [reply controllers/sriovnetworknodepolicy_controller.go:174]: +1

a deleted policy also affects node states spec

**Comment by adrianchiris** [comment]
>general question is it possible for the user to create an object with that name?
>
>can we use a non existing namespace or something like that to be sure there is not going to be an object like that?

why do we care ? its just an "event" to trigger reconcile that is not dependent on the specific obj that triggered the event.

File: controllers/sriovnetworknodepolicy_controller.go (hunk 2 above)
Comment (zeeke) [reply controllers/sriovnetworknodepolicy_controller.go:174]: good point! fixing

File: controllers/sriovnetworknodepolicy_controller.go (hunk 3 above)
Comment (zeeke) [reply controllers/sriovnetworknodepolicy_controller.go]: done

**Comment by zeeke** [comment]
> general question is it possible for the user to create an object with that name?
>
> can we use a non existing namespace or something like that to be sure there is not going to be an object like that?

Good point. If the user creates a node policy called `node-policy-sync-event`, then the Reconcile logic is invoked twice. Not a big deal, but this PR is all about avoiding doing the same things multiple times. Changing.


> why do we care ? its just an "event" to trigger reconcile that is not dependent on the specific obj that triggered the event.

Because the controller (and the Reconcile method) are still triggered by every SriovNetworkNodePolicy, as it's the `.For(...)` argument.

**Review by SchSeba — APPROVED** [review]
LGTM

thanks!

**Pull request merged**